modules/apps/info/info-impl/src/main/java/com/liferay/info/internal/item/InfoItemServiceTrackerImpl.java

```diff
@@ -72,28 +78,50 @@
 
 	@Override
 	public <P> List<P> getAllInfoItemServices(
-		Class<P> serviceClass, String itemClassName) {
-
-		ServiceTrackerMap<String, List<P>> infoItemServiceTrackerMap =
-			(ServiceTrackerMap<String, List<P>>)
-				_itemClassNameInfoItemServiceTrackerMap.get(
-					serviceClass.getName());
-
-		List<P> services = infoItemServiceTrackerMap.getService(itemClassName);
+		Class<P> serviceClass, String itemClassName, String filterString) {
+
+		ServiceTrackerMap
+			<String, ? extends List<ServiceReferenceServiceTuple<P, P>>>
+				infoItemServiceTrackerMap =
+					(ServiceTrackerMap)
+						_itemClassNameInfoItemServiceTrackerMap.get(
+							serviceClass.getName());
+
+		List<ServiceReferenceServiceTuple<P, P>> serviceReferenceServiceTuples =
+			infoItemServiceTrackerMap.getService(itemClassName);
+
+		if (serviceReferenceServiceTuples != null) {
+			Stream<ServiceReferenceServiceTuple<P, P>> stream =
+				serviceReferenceServiceTuples.stream();
+
+			if (filterString != null) {
+				try {
+					Filter filter = FrameworkUtil.createFilter(filterString);
+
+					stream = stream.filter(
+						srst -> filter.match(srst.getServiceReference()));
+				}
+				catch (InvalidSyntaxException invalidSyntaxException) {
```

Comment:
I think it's best to throw this, since it should only happen when there is a developer error and in that case the sooner the developer knows about the error, the better.
What do you think?

Reply:
yup, I agree... or at least log it...